api/src/client/dynamics/DynamicsLicenseStatusClient.ts

```diff
@@ -0,0 +1,71 @@
+import { SearchLicenseStatus } from "@domain/types";
+import { LogWriterType } from "@libs/logWriter";
+import { LicenseSearchNameAndAddress, LicenseStatusResult } from "@shared/license";
+import { searchBusinessAddressesForMatch } from "./searchBusinessAddressesForMatch";
+import {
+  AccessTokenClient,
+  BusinessAddressClient,
+  BusinessIdClient,
+  ChecklistItemsClient,
+  LicenseApplicationIdClient,
+} from "./types";
+
+type Config = {
+  tokenFactory: AccessTokenClient;
+  businessIdFactory: BusinessIdClient;
+  businessAddressFactory: BusinessAddressClient;
+  applicationIdFactory: LicenseApplicationIdClient;
+  checkListItemFactory: ChecklistItemsClient;
```

Comment:
[sand] This is a little nitpicky, but I wouldn't name these variables with the word `Factory`. (and when you _do_ that, they should reflect what is actually being returned)

In my opinion, and in the patterns we use, a "ThingFactory" is a function that returns a Thing. 

So a `checklistItemsFactory` by naming pattern would mean "a function that returns to me checklist items" - which doesn't make sense, because traditionally a "factory" returns an object, not a piece of data.

Maybe the following naming / type matchup will help explain:
```
checklistItemsFactory: () => ChecklistItems[] // as stated above, doesn't really make sense
checklistItemsClientFactory: () => ChecklistItemsClient // this is an expected use of a factory
checklistItemsClient: ChecklistItemsClient // when you just have an object of that type, it's not a factory, so the variable name should reflect that
```